### PR TITLE
test(eo-runtime): reproduce bug EO app calls varargs func

### PR DESCRIPTION
**Pull request by baudoliver7**
EO app fails with an exception `You can't overwrite X` when it tries to call a varargs function. The error occurs within varargs function during its dataization. We reproduced it here by Java and EO test and introduced a todo to fix it.

Close: #414

### CODE REVIEW

**Comment by baudoliver7** [comment]
@yegor256 please, check the test that reproduces the bug and merge if all things are right.

File: eo-runtime/src/test/java/org/eolang/DataizedTest.java [hunk 1]
```diff
@@ -76,4 +78,102 @@ public void close() throws SecurityException {
...
+        public EOappCallingVarargsFunc(Phi sigma) {
+            super(sigma);
+            this.add(
+                "φ",
+                new AtOnce(
+                    new AtComposite(
+                        this,
+                        (rho) -> {
+                            Phi fvar = new PhCopy(new DataizedTest.Fvarargs(rho));
+                            Phi var1 = new PhWith(
```
Comment (yegor256) [inline eo-runtime/src/test/java/org/eolang/DataizedTest.java:117]: @baudoliver7 you can do it shorter, just like this: `Phi var1 = new Data.ToPhi(1L)`

**Comment by yegor256** [comment]
@baudoliver7 see one comment above, you can make your tests much more compact. Also, I would suggest reproducing it FIRST in `runtime-tests.eo` with EO code. Then, with Java.

File: eo-runtime/src/test/java/org/eolang/DataizedTest.java (hunk 1 above)
Comment (baudoliver7) [reply eo-runtime/src/test/java/org/eolang/DataizedTest.java:117]: @yegor256 Ah, I see

**Comment by baudoliver7** [comment]
> @baudoliver7 see one comment above, you can make your tests much more compact. Also, I would suggest reproducing it FIRST in `runtime-tests.eo` with EO code. Then, with Java.

@yegor256 How could I execute an EO test ?

**Comment by baudoliver7** [comment]
@yegor256 I see now. EO tests execute automatically :-)
I uncomment the test since it is not yet resolved.

**Comment by baudoliver7** [comment]
@yegor256 please, see latest changes.

File: eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                    )
+                )
+            );
+        }
+    }
+
+    /**
+     * Varargs function sample.
+     * {@code
+     *  [args] > f
```
Comment (yegor256) [inline eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java:102]: @baudoliver7 I think here should be:

```
[args...] > f
```

right?

File: eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        public EOappThatCallsVarargsFunc(Phi sigma) {
+            super(sigma);
+            this.add(
+                "φ",
+                new AtOnce(
+                    new AtComposite(
+                        this,
+                        (rho) -> {
+                            Phi fvar = new PhCopy(new Fvarargs(rho));
+                            Phi var1 = new Data.ToPhi(1L);
```
Comment (yegor256) [inline eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java:83]: @baudoliver7 what's the point of making these variables? Can't you just inline values where they are used?

**Comment by yegor256** [comment]
@baudoliver7 see comments above. You are using too many variables. Read this: https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html

File: eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java (hunk 2 above)
Comment (baudoliver7) [reply eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java:102]: Well seen !

File: eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java (hunk 3 above)
Comment (baudoliver7) [reply eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java:83]: Got it :)

**Comment by baudoliver7** [comment]
@yegor256 please, see my latest changes.

File: eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+     *    2 > @
+     * }
+     * @since 0.22
+     */
+    @SuppressWarnings("unchecked")
+    private static class Fvarargs extends PhDefault {
+        public Fvarargs(final Phi sigma) {
+            super(sigma);
+            this.add("args", new AtVararg());
+            this.add(
```
Comment (yegor256) [inline eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java:110]: @baudoliver7 why do you need this attribute here? It doesn't seem to be relevant to tests... what it's for?

**Comment by yegor256** [comment]
@baudoliver7 see above

File: eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java (hunk 4 above)
Comment (baudoliver7) [reply eo-runtime/src/test/java/EOorg/EOeolang/EODataizeAppCallsVarargsFuncTest.java:110]: @yegor256 it's to reproduce the function :
```eo
[args...] > f
  1 > a
  2 > @
```
The error appears only when there is another attribute. It vanishes when we write :
```eo
[args...] > f
  2 > @
```

**Comment by yegor256** [comment]
@rultor merge
